releasenotes/notes/add-hls-plugins-038388970ad43c55.yaml

```diff
@@ -0,0 +1,14 @@
+---
+features:
+  - |
+    Added high-level-synthesis plugins for :class:`.LinearFunction` and for
+    :class:`qiskit.quantum_info.Clifford`, extending the set of synthesis
+    methods that can be called from :class:`~qiskit.transpiler.passes.HighLevelSynthesis`
+    transpiler pass.
+    For :class:`.LinearFunction` the newly available plugins are
+    :class:`.PMHSynthesisLinearFunction` and :class:`.KMSSynthesisLinearFunction`.
+    For :class:`qiskit.quantum_info.Clifford` the newly available plugins are
+    :class:`.AGSynthesisClifford`, :class:`.BMSynthesisClifford`,
+    :class:`.GreedySynthesisClifford` and :class:`.LayerSynthesisClifford`.
+    Please refer to :mod:`qiskit.synthesis` documentation for more information
+    about each individual method.
```

Comment:
Can you add a table or something that maps the plugin names to the methods? I think that'll be good for users so they know what the plugin names are going to be. Also having an example of how to use it with transpile would go a long way too.

Reply:
Done in [0dbcf31](https://github.com/Qiskit/qiskit-terra/pull/9399/commits/0dbcf319dc2cbcd43c75a2c13fae534adddac1e7).